src/Symfony/Component/VarDumper/Dumper/HtmlDumper.php

```diff
@@ -184,6 +223,88 @@ function toggle(a, recursive) {
     return true;
 };
 
+function collapse(a, recursive) {
+    var s = a.nextSibling || {}, oldClass = s.className;
+
+    if ('sf-dump-expanded' == oldClass) {
+        toggle(a, recursive);
+
+        return true;
+    }
+
+    return false;
+};
+
+function expand(a, recursive) {
+    var s = a.nextSibling || {}, oldClass = s.className;
+
+    if ('sf-dump-compact' == oldClass) {
+        toggle(a, recursive);
+
+        return true;
+    }
+
+    return false;
+};
+
+function collapseAll(root)
+{
+    var a = root.querySelector('a.sf-dump-toggle');
+    if (a) {
+        collapse(a, true);
+        expand(a);
+
+        return true;
+    }
+
+    return false;
+}
+
+function reveal(node) {
+    var current = node;
+    var parents = [];
+    var previous = null;
+    var parent = null;
+
+    while ((parent = current.parentNode || {}) && (previous = parent.previousSibling)) {
+        current = current.parentNode;
+        parents.push(previous);
+    }
+
+    if (0 !== parents.length) {
+        parents.forEach(function (parent) {
```

Comment:
Unfortunately it won't, because the index will be passed as second argument of `expand`, which expects a boolean. The index would wrongly be interpreted as `true`.